pkg/operator/controller/controller.go

```diff
@@ -101,64 +100,56 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			// resource that got deleted async).
 			log.Info("clusteringress not found; reconciliation will be skipped", "request", request)
 		} else {
-			// Try again later.
-			log.Error(err, "failed to get clusteringress", "request", request)
-			result.RequeueAfter = 10 * time.Second
+			errs = append(errs, fmt.Errorf("failed to get clusteringress %q: %v", request, err))
 		}
 		ingress = nil
 	}
 
 	caSecret, err := r.ensureRouterCACertificateSecret()
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to ensure CA secret: %v", err)
+		errs = append(errs, fmt.Errorf("failed to ensure CA secret: %v", err))
+		caSecret = nil
```

Comment:
Throughout: if `err != nil` then we can't rely on the result of the function (`caSecret` in this case).

See https://github.com/openshift/cluster-ingress-operator/pull/136 for an approach which avoids this

Reply:
And also `caSecret = nil` is us basically making up a result for the function which returned an error; the actual result is "don't know if caSecret is nil"

Reply:
I don't think of `caSecret == nil` as meaning necessarily that we know there is no secret; rather, it means the input is unavailable.